crypto/hpke/hpke.c

```diff
@@ -292,25 +308,322 @@ static int x25519_auth_decap(const EVP_HPKE_KEY *key,
   return 1;
 }
 
+static const EVP_HPKE_KEM hpke_kem_x25519_hkdf_sha256 = {
+    /*id=*/EVP_HPKE_DHKEM_X25519_HKDF_SHA256,
+    /*internal_kem_nid=*/NID_HPKE_DHKEM_X25519_HKDF_SHA256,
+    /*internal_digest_nid=*/NID_sha256,
+    /*is_authenticated=*/true,
+    /*public_key_len=*/X25519_PUBLIC_VALUE_LEN,
+    /*private_key_len=*/X25519_PRIVATE_KEY_LEN,
+    /*seed_len=*/X25519_PRIVATE_KEY_LEN,
+    /*enc_len=*/X25519_PUBLIC_VALUE_LEN,
+    /*shared_secret_len=*/X25519_SHARED_KEY_LEN,
+    x25519_init_key,
+    x25519_generate_key,
+    x25519_encap_with_seed,
+    x25519_decap,
+    x25519_auth_encap_with_seed,
+    x25519_auth_decap,
+    X25519_public_from_private,
+};
+
 const EVP_HPKE_KEM *EVP_hpke_x25519_hkdf_sha256(void) {
-  static const EVP_HPKE_KEM kKEM = {
-      /*id=*/EVP_HPKE_DHKEM_X25519_HKDF_SHA256,
-      /*public_key_len=*/X25519_PUBLIC_VALUE_LEN,
-      /*private_key_len=*/X25519_PRIVATE_KEY_LEN,
-      /*seed_len=*/X25519_PRIVATE_KEY_LEN,
-      /*enc_len=*/X25519_PUBLIC_VALUE_LEN,
-      x25519_init_key,
-      x25519_generate_key,
-      x25519_encap_with_seed,
-      x25519_decap,
-      x25519_auth_encap_with_seed,
-      x25519_auth_decap,
-  };
-  return &kKEM;
+  return &hpke_kem_x25519_hkdf_sha256;
+}
+
+// Secret Key Surgery
+// These stunts are performed by trained professionals, do not try this at home.
+// FIXME: This should ideally be a part of the KEM interface.
```

Comment:
you are absolutely right, I wish we had all these things nicely defined and implemented at the KEM layer -- I hope I'll get to it soon!